File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
          int n = nums.length;

        Map<Integer, Integer> sumPrefixMap = new HashMap<>();
        sumPrefixMap.put(0, -1);
        int sum = 0;

        for (int i = 0; i <n; i++) {
            sum += nums[i];
            if (k != 0) {
                sum = sum % k;
            }
            if (sumPrefixMap.containsKey(sum)) {
                if(i - sumPrefixMap.get(sum) > 1) {
                    return true;
                }
            }else {
                sumPrefixMap.put(sum, i);
            }

        }
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.ts
function checkSubarraySum(nums: number[], k: number): boolean {

    const map = new Map();
    map.set(0, -1);
    let sum = 0;
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
        if (k !== 0) {
            sum %= k;
        }
        if (map.has(sum)) {
            if (i - map.get(sum) > 1) {
                return true;
            }
        } else {
            map.set(sum, i);
        }
    }
    return false;
};